### PR TITLE
Orthoview support

### PR DESCRIPTION
**Pull request by dchristiaens**
This PR adds support for showing three orthogonal slices in horizontal lay-out, as suggested in #17. Orthoview is toggled with `o`; `s / c / a` continue to select the "active" plane to control the focus. The slice and crosshairs are updated in all planes.

Screenshot 2020-06-20 at 20 22 12

As you can see, panels with shorter dimensions are currently mapped to the top left corner. Is it worth trying to align these to centre? The code may also need some cleaning up...

### CODE REVIEW

**Comment by jdtournier** [comment]
Well that was quick! Nice work!

> Is it worth trying to align these to centre?

Not at this stage. Leave it to others to implement if it really offends their sensitivities...

**Comment by jdtournier** [comment]
Oops! Sorry, slip of the finger...

> The code may also need some cleaning up...

This is the bit I'd focus on before merging. It's worth making sure the backend is clean and easy to work with. But again, not critical before merge - can always be done later.

**Comment by dchristiaens** [comment]
> It's worth making sure the backend is clean and easy to work with. But again, not critical before merge - can always be done later.

I think it's good enough for now. The backend is clean; it's `cmd/mrpeek.cpp` that I think is getting a bit long... I would like to avoid the code duplication between [orthoview](https://github.com/MRtrix3/mrpeek/blob/c890aef52b52a2f6114c668a5bd2ba64147a110d/cmd/mrpeek.cpp#L232-L252) and [regular view](https://github.com/MRtrix3/mrpeek/blob/c890aef52b52a2f6114c668a5bd2ba64147a110d/cmd/mrpeek.cpp#L262-L276) but I run into a segfault that I can trace back to the adapter but can't seem to resolve.

Unless there are issues with this version, I think this can be merged to master.

**Comment by dchristiaens** [comment]
> > Is it worth trying to align these to centre?
> 
> Not at this stage. Leave it to others to implement if it really offends their sensitivities...

It was offending mine 😄, and it turned out to be only 4 lines of code anyways...

**Comment by maxpietsch** [comment]
Good to go from my side. I added bounding boxes and highlight that of the current view plane which should aid in figuring out to which view plane the focus corresponds. For this, I added two more colours to the ColourMap.

image

image

Also added command line and config file options to toggle orthoview.

**Review by jdtournier — APPROVED** [review]
Looks great guys!

**Comment by dchristiaens** [comment]
The details make all the difference. Well done Max!

**Pull request merged**
